### DISCUSSION TITLE
support other Format Files

### DISCUSSION THREAD

**Discussion by sh4m2hwz** [fkie-cad/cwe_checker#253] [General]
you are planning to do support for PE, DEX, MATH-O, COFF, BINARY?
if not, what pieces of code are responsible for mapping the binar and extracting instructions?
which interfaces can be overridden, responsible for working with the file format?

**Comment by Enkelmann** [comment]
Moved to discussions, since it is more a general question than a concrete issue. But feel free to open issues for particular binary formats!

Right now we support ELF files and we have experimental (and at the moment poorly tested) support for PE files and binaries of bare-metal firmware samples. In the far future we would like to support as many binary file formats as Ghidra supports. But this is not a priority until we are satisfied with the analysis quality for ELF files or until we find a sponsor for the work on this particular aspect of the cwe_checker.

For supporting a new binary format one has to add parsing capabilities for in in the `RuntimeMemoryImage` struct (located in `src/cwe_checker_lib/src/utils/binary.rs`). This should in theory already enable analysis of the file format. But then one has to debug the analysis to find out whether the implementation of the analysis uses some implicit assumptions that differ for binaries of the new format. For example, PE files use different calling conventions that ELF files, which must be considered by the analysis (see issue #250).
